components/forms/select-input/select-input.js

```diff
@@ -0,0 +1,134 @@
+function SelectInput(id, options, validationMsg, placeholder, isRequired) {
+  let selectInput
+  let selectElement
+  let validationMsgElement
+  let isOpen = false
+  let displayValidationMsg = false
+  const _options = options
+  const _validationMsg = validationMsg
+  const _placeholder = isRequired ? placeholder + '*' : placeholder
+  let _valid = false
+  let _value = 'not set'
+  let _index = 0
+  let shouldPopulateList = true
+
+  const cacheDom = function() {
+    selectInput = document.getElementById(id)
+    selectElement = selectInput.querySelector('.msds-select-input__select')
+    validationMsgElement = selectInput.querySelector('.msds-select-input__validation-msg')
+  }
+
+  const bindEvents = function() {
+    selectElement.addEventListener('change', () => render())
+  }
+
+  const render = function() {
+    _index = selectElement.value
+    validationMsgElement.innerHTML = _validationMsg
+    if (isOpen) {
+      selectInput.classList.toggle('msds-select-input--open')
+    }
+    if (_validationMsg != '' && displayValidationMsg == true) {
```

Comment:
agree, because:
1. to be consistent, in another cases in render function all another elements add/remove css clases
2. using inline styles are not recomended:
- they are agressive, not possible to overwrite them
- when styles are set by js, it's more difficult to debug them since html element has different styles then defined in css file
- doesnt follow good convention of seperation of contexts (css-styles, js-action, html-content)

Reply:
ive updated it ot use a modifier instead